can/broadcastmanager.py

```diff
@@ -71,44 +110,72 @@ def start(self):
 class ModifiableCyclicTaskABC(CyclicSendTaskABC):
     """Adds support for modifying a periodic message"""
 
-    def modify_data(self, message):
-        """Update the contents of this periodically sent message without altering
-        the timing.
+    def _check_modified_messages(self, messages):
+        """Helper function to perform error checking when modifying the data in
+        the cyclic task.
 
-        :param can.Message message:
-          The message with the new :attr:`can.Message.data`.
-          Note: The arbitration ID cannot be changed.
+        Performs error checking to ensure the arbitration ID and the number of
+        cyclic messages hasn't changed.
+
+        Should be called when modify_data is called in the cyclic task.
+        """
+        if len(self.messages) != len(messages):
+            raise ValueError(
+                "The number of new cyclic messages to be sent must be equal to "
+                "the number of messages originally specified for this task"
+            )
+        if self.arbitration_id != messages[0].arbitration_id:
+            raise ValueError(
+                "The arbitration ID of new cyclic messages cannot be changed "
+                "from when the task was created"
+            )
+        return messages
+
+    def modify_data(self, messages):
+        """Update the contents of the periodically sent messages, without
+        altering the timing.
+
+        :param Union[Sequence[can.Message], can.Message] messages:
+            The messages with the new :attr:`can.Message.data`.
+
+            Note: The arbitration ID cannot be changed.
+
+            Note: The number of new cyclic messages to be sent must be equal
+            to the original number of messages originally specified for this
+            task.
         """
-        self.message = message
+        messages = self._check_and_convert_messages(messages)
+        messages = self._check_modified_messages(messages)
```

Comment:
Minor, but `self._check_modified_messages(messages)` do not make any modification to messages, so you do not really need to return `messages`